drivers/clock_control/Kconfig.wch_rcc

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2024 Michael Hope
+# SPDX-License-Identifier: Apache-2.0
+
+config CLOCK_CONTROL_WCH_RCC
+	bool "WCH_RCC clock control"
+	default y
+	depends on DT_HAS_WCH_RCC_ENABLED
+	help
+	  Enable driver for RCC
+
+config WCH_CH32V00X_HSI
+	bool "Internal 24 MHz high speed RC oscillator (HSI)"
+	help
+	  Enable the internal 24 MHz RC oscillator at startup.
+	  This can then be selected as the main clock reference for the SOC.
+
+config WCH_CH32V00X_HSE
+	bool "External 4..25 MHz oscillator (HSE)"
+	default y
+	help
+	  Enable the high speed external oscillator at startup.
+	  This can then be selected as the main clock reference for the SOC.
+
+config WCH_CH32V00X_LSI
+	bool "Internal 128 kHz low speed RC oscillator (LSI)"
+	help
+	  Enable the internal 128 kHz low speed oscillator at startup.
+
+config WCH_CH32V00X_PLL
+	bool "Internal PLL"
+	default y
+	help
+	  Enable the internal PLL at startup.
+	  This can then be selected as the main clock reference for the SOC.
+
+choice
+	prompt "PLL clock source"
+	default WCH_CH32V00X_HSE_AS_PLLSRC
+	help
+	  Selects which clock to use as PLL clock.
+
+config WCH_CH32V00X_DEFAULT_AS_PLLSRC
+	bool "DEFAULT"
+	help
+	  This choice will leave the PLL source in its initial state.
+	  This can be the default reset state, i.e. HSI as PLLSRC, or a
+	  state set by a bootloader.
+
+config WCH_CH32V00X_HSI_AS_PLLSRC
+	bool "HSI"
+	depends on WCH_CH32V00X_HSI
+
+config WCH_CH32V00X_HSE_AS_PLLSRC
+	bool "HSE"
+	depends on WCH_CH32V00X_HSE
+
+endchoice
+
+choice
+	prompt "Main clock reference"
+	default WCH_CH32V00X_PLL_AS_SYSCLK
+	help
+	  Selects which clock to use as the main SYSCLK. Other clocks, such
+	  as HCLK, will be derived from SYSCLK.
+
+config WCH_CH32V00X_DEFAULT_AS_SYSCLK
+	bool "DEFAULT"
+	help
+	  This choice will leave all clocks in their initial state.
+	  This can be the default reset state, i.e. HSI with no PLL, or a
+	  state set by a bootloader.
+
+config WCH_CH32V00X_PLL_AS_SYSCLK
+	bool "PLL"
+	depends on WCH_CH32V00X_PLL
+
+config WCH_CH32V00X_HSI_AS_SYSCLK
+	bool "HSI"
+	depends on WCH_CH32V00X_HSI
+
+config WCH_CH32V00X_HSE_AS_SYSCLK
+	bool "HSE"
+	depends on WCH_CH32V00X_HSE
+
+endchoice
```

Comment:
-1: please, use DT for hw configuration. At this point I don't think it is acceptable anymore, and specially here where ST/GD do it properly already and this SoC is "quite similar".

Reply:
@kholia I'll have a look at this and send you a PR

Reply:
Thank you! I was feeling a bit lost here.

Some of the review comments conflict with each other.

Reply:
https://github.com/nzmichaelh/zephyr/commit/698fe5b5a3fba21e3bd3e514dd9c7f14240a85ee moves the clock tree to Devicetree.

Reply:
Applied and pushed - thanks!